### PR TITLE
108 services

### PR DESCRIPTION
**Pull request by HadarRose**
- Created services: TwitterCreationService, TwitterExceptionHandlerService, TwitterResourceService, ConfigService
- Removed service functionalities from original classes 
- Refactored ExceptionHandler to be TwitterExceptionHandlerService
- Updated tests to fit new architecture

### CODE REVIEW

**Review by kev-chan — CHANGES_REQUESTED** [review]
I want you to think about how others may want to reuse your service, namely TwitterService.  Are there any assumptions you are making and if so, please make changes.  If not, let's talk when you are out of ideas.

**Comment by kev-chan** [comment]
for more context, other developers are code.  Why separate out between Resources and Services in the first place?

**Comment by HadarRose** [comment]
@kev-chan in response to your comments: I think I understand what you mean. If I understand correctly, the point of the refactoring should have been to allow TwitterRequestResource to be independent of the way we choose to reach Twitter.  Meaning that if we chose to make our own calls to Twitter API w/o twitter4j, for example, all we'd have to change is the service, not both the service and the resource. I'll make the Resource class completely independent of twitter4j.

File: src/main/java/configuration/ApplicationConfiguration.java [hunk 1]
```diff
@@ -1,74 +1,25 @@
...
 import io.dropwizard.Configuration;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-import twitter4j.Twitter;
-import twitter4j.TwitterFactory;
-import twitter4j.conf.ConfigurationBuilder;
-
-import java.io.File;
-import java.io.IOException;
+import model.TwitterKeys;
```
Comment (kev-chan) [inline src/main/java/configuration/ApplicationConfiguration.java]: package you typically want <something>.<somethig> and not just model.TwitterKeys.  The extended package makes it much less likely to clash with other dependencies.  This applies to all your code.

File: src/main/java/model/TwitterKeys.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
+package model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class TwitterKeys {
```
Comment (kev-chan) [inline src/main/java/model/TwitterKeys.java]: why the model package? when this is an extension of the configuration

File: src/test/java/services/twitter4j/TwitterResourceExceptionTest.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        when(exception.getErrorCode()).thenReturn(2);
+
+        TwitterResourceException twitterResourceException = new TwitterResourceException(exception);
+        assertEquals(twitterResourceException.GENERAL_ERROR, twitterResourceException.getMessage());
+        assertEquals(403, twitterResourceException.getStatusCode());
+    }
+
+    @Test
+    public void tweetLengthError() {
+        TwitterException exception = mock(TwitterException.class);
```
Comment (kev-chan) [inline src/test/java/services/twitter4j/TwitterResourceExceptionTest.java:35]: for future reference but no need to change now
you could probably simplify with the following
TwitterExceotion exception = new TwitterException();
exception.setStatusCode(...)
it's simpler w/o the mocks and you've got this pattern in multiple places

File: src/test/java/resources/TwitterRequestResourceTest.java [hunk 4]
```diff
@@ -1,102 +1,84 @@
...
 public class TwitterRequestResourceTest {
-    private Twitter mockedTwitter;
-    private ExceptionHandler mockedHandler;
+    private TwitterResourceService mockedResourceServices;
 
     @Before
     public void initialize() {
-        mockedTwitter = mock(Twitter.class);
-        mockedHandler = mock(ExceptionHandler.class);
+        mockedResourceServices = mock(TwitterResourceService.class);
```
Comment (kev-chan) [inline src/test/java/resources/TwitterRequestResourceTest.java:23]: good use of mock

File: src/main/java/services/twitter4j/TwitterResourceService.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    private Twitter twitter;
+
+    /**
+     * Constructor
+     *
+     * @param twitterKeys TwitterKeys containing keys for new twitter object
+     */
+    public TwitterResourceService(TwitterKeys twitterKeys) {
+        logger.info("TwitterResourceService created");
+        TwitterCreationService twitterCreationService = new TwitterCreationService(twitterKeys);
```
Comment (kev-chan) [inline src/main/java/services/twitter4j/TwitterResourceService.java]: do you actually need a twitter creation service?

**Review by kev-chan — CHANGES_REQUESTED** [review]

**Pull request merged**